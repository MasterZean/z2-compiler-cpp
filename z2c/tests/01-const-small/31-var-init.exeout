var a: Small = Small{0}
var b: Small = Small{-128}
var c: Small = Small{127}
var h: Small = Small{0}
var j: Small = Small{97}
var k: Small = Small{127}
