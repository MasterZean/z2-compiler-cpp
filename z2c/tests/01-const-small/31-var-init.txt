val a: Small = 0
val b: Small = -128
val c: Small = 127
val d: Small = -129
val e: Small = 128
val f: Small = 0.0f
val g: Small = 0.0
val h: Small = 0p
val i: Small = 128p
val j: Small = 'a'
val k: Small = '\u00007F'
val l: Small = '\u000080'
val m: Small = false
val n: Small = true
