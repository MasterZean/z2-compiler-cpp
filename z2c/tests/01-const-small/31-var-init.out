int8 a = 0;
int8 b = -128;
int8 c = 127;
int8 h = 0u;
int8 j = (int8)('a');
int8 k = (int8)(127);
DummyClass(4, 14): error:
	can't assign 'Int' instance to 'Small' instance
DummyClass(5, 14): error:
	can't assign 'Int' instance to 'Small' instance
DummyClass(6, 14): error:
	can't assign 'Float' instance to 'Small' instance
DummyClass(7, 14): error:
	can't assign 'Double' instance to 'Small' instance
DummyClass(9, 14): error:
	can't assign 'PtrSize' instance to 'Small' instance
DummyClass(12, 14): error:
	can't assign 'Char' instance to 'Small' instance
DummyClass(13, 14): error:
	can't assign 'Bool' instance to 'Small' instance
DummyClass(14, 14): error:
	can't assign 'Bool' instance to 'Small' instance
